server/controllers/user-controller.js

```diff
@@ -0,0 +1,80 @@
+import {
+  successResponse,
+  errorResponse,
+  validate,
+  validationErrorResponse,
+  checkUniqueUserName
+} from '../utils/helpers';
+import db from '../models';
+import { profileSchema } from '../utils/validators';
+
+const { User } = db;
+
+/**
+ * The controllers for users route
+ *
+ * @class UsersController
+ */
+class UsersController {
+  /**
+   * update user profile
+   * @static
+   * @param {Request} req request object
+   * @param {Response} res response object
+   * @memberof {Users}
+   * @returns {undefined}
+   */
+  static async updateUser(req, res) {
+    validate(req.body, profileSchema)
+      .then(async () => {
+        const { ...auth } = req.authUser;
+        try {
+          const { username, bio } = req.body;
+          const user = await User.findOne({
+            where: { id: auth.id }
+          });
+          const isUnique = await checkUniqueUserName(username, auth.id);
+          if (!isUnique) {
+            return errorResponse(res, 'username already exist', 409);
+          }
+          const data = await user.update(
+            { username, bio },
+            { returning: true, where: { id: auth.id } }
+          );
+          successResponse(
+            res,
+            { message: 'update successful', user: data },
+            200
+          );
+        } catch (error) {
+          errorResponse(res, error.message, 500);
+        }
+      })
+      .catch(({ details }) => {
+        validationErrorResponse(res, details, 400);
+      });
+  }
+
+  /**
+   * update user profile
+   * @static
+   * @param {Request} req request object
+   * @param {Response} res response object
+   * @memberof {Users}
+   * @returns {undefined}
+   */
+  static async getProfile(req, res) {
+    try {
+      const { id } = req.authUser;
+      const user = await User.findOne({ where: { id } });
+      if (user) {
+        delete user.hash;
+        successResponse(res, { data: user }, 200);
```

Comment:
The response object should be `user` and not `data` according to the API spec

Reply:
i will work on this